sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/CosmosConfig.scala

```diff
@@ -537,25 +537,25 @@ private object CosmosAuthConfig {
         parseFromStringFunction = authTypeAsString =>
             CosmosConfigEntry.parseEnumeration(authTypeAsString, CosmosAuthType),
         helpMessage = "There are two auth types are supported currently: " +
-            "`MasterKey`(PrimaryReadWriteKeys, SecondReadWriteKeys, PrimaryReadOnlyKeys, SecondReadWriteKeys), `ServicePrinciple`")
+            "`MasterKey`(PrimaryReadWriteKeys, SecondReadWriteKeys, PrimaryReadOnlyKeys, SecondReadWriteKeys), `ServicePrincipal`")
```

Comment:
For backwards compatibility the func would also need to keep working for the incorrectly spelled `ServicePrinciple`

Reply:
@FabianMeiswinkel For the pull request being submitted, the new version would now just not know what the old spelling was, so I am struggling to find a decent way of doing this that would not be a breaking changes that wouldn't involve implementing both spellings as 2 valid spellings, to actually fix it and rename it, I feel like it has to be a breaking change rather than attempt at making a non breaking change but creating confusion and complexity for no gain?